controllers/vspherevm_controller.go

```diff
@@ -100,6 +100,11 @@ func (r vmReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr error)
 		return reconcile.Result{}, err
 	}
 
+	if vsphereVM.Status.FailureReason != nil || vsphereVM.Status.FailureMessage != nil {
```

Comment:
Having this here will prevent the deletion from proceeding (to eventually remove the finalizer) - won't it?

Reply:
+1 this should be in `reconcileNormal`